src/lib.rs

```diff
@@ -142,6 +142,22 @@ fn decode_hex_nibble(n: u8x16) -> Result<u8x16, Error> {
     Ok(result)
 }
 
+#[inline]
+fn nibble_chunck(chunk: &[u8]) -> (u8x16, u8x16) {
+    let parsed_chunk = u8x32::from_slice(chunk);
+    let mut high_bytes_vec: Vec<u8> = vec![]; 
```

Comment:
Instead of using vec to allocate on the heap, can we just set `[u8; X]` directly? that way we don't need to allocate memory.